datamade/just-spaces
# Create widget for defining CensusAreas

## Overview

Closes #101, closes #158 

Define a `MultiSelectGeometryWidget` for defining CensusAreas using Census block groups.

In addition, update the import pipeline to:

1. Import Census block group shapefiles for Philadelphia
2. Make the import of ACS data accommodate data refreshes

## Testing Instructions

### Setup

- Run migrations

### Test the Makefile changes

- Remove the derived shapefile: `rm data/final/shapefiles/cb_2018_42_bg_500k.shp`
- Change to the data directory: `cd data`
- Recreate the file: `docker-compose run --rm make final/shapefiles/cb_2018_42_bg_500k.shp`

### Test the data import

- Run `python manage.py import_data` and confirm that:
    - Your existing ACS data gets updated, not deleted (the script should print to the console indicating this)
    - The script successfully imports blockgroup data

### Test the multiselect widget

- Run the development server
- Navigate to http://localhost:8000/census-areas/create/ and try creating CensusAreas:
    - Create a CensusArea with one block group
    - Create a CensusArea with multiple block groups
    - Try to create a CensusArea with no block groups, and confirm that the form shows you an error
    - After creating CensusAreas, view their "Edit" pages and confirm that the selected block groups reappear on the map
- Navigate to the DAD for an ACS-compatible survey and try using your new CensusAreas to compare data
    - Confirm that the bars look different across CensusAreas
